### PR TITLE
Outcome Prediction Experimentatinon Pipeline

### PR DESCRIPTION
**Pull request by Benny-Gil**
This PR introduces a **Centralized Experiment Pipeline** that refactors the previous scattered Jupyter Notebooks into a streamlined, reproducible Python workflow. This pipeline handles data preprocessing, various resampling techniques, model training, and automatically exports results directly to LaTeX-compatible tables.

This also updates multiple-author citations for APA 7 compliance, fixes unknown author citation, and updates irrelevant references. 

### Centralized Experiment Pipeline
```mermaid
graph TD
    A[Raw Data] --> B[Data Preprocessing & Imputation]
    B --> C{Resampling Strategy}
    C -->|Baseline| D[No Resampling]
    C -->|SMOTE| E[SMOTE]
    C -->|SMOTE-ENN| F[SMOTE-ENN]
    C -->|SMOTE-Tomek| G[SMOTE-Tomek]
    
    D --> H[Model Training & Cross-Validation]
    E --> H
    F --> H
    G --> H
    
    H --> I[Evaluation Metrics Calculation]
    I --> J[Visualization Generation]
    I --> K[LaTeX Tables Export]
    I --> L[Model Serialization .onnx]
```

### CODE REVIEW

File: non-temporal/experiment_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    def load_data(self, file_path):
+        df_raw = pd.read_csv(file_path)
+        
+        # Decode Outcome/Status
+        OUTCOME_LABELS_ALPHA = [
+            'CURED', 'DIAGNOSED', 'DIED', 'EXCLUDED', 'FAILED', 'LOST TO FF-UP',
+            'NOT ENROLLED', 'NOT EVALUATED', 'ON TREATMENT', 'Screened', 'TREATMENT COMPLETED'
+        ]
+        outcome_encoded_sorted = sorted(df_raw['Outcome/Status'].dropna().unique())
+        OUTCOME_DECODE = dict(zip(outcome_encoded_sorted, OUTCOME_LABELS_ALPHA))
```
Comment (VladTemp27) [inline non-temporal/experiment_pipeline.py:64]: The code assumes sorted encoded values will always align 1:1 with alphabetically sorted labels (defined at line 51-54). If the dataset adds/removes an outcome category, the mapping silently shifts, producing wrong targets and an invalid model. No validation exists.

Fix: Use explicit mapping (encoded_value → label) or add assertion: `assert len(outcome_encoded_sorted) == len(OUTCOME_LABELS_ALPHA)`.

File: paper/apa/tables/baseline_results.tex [hunk 2]
```diff
@@ -0,0 +1,19 @@
+\begin{table}[htbp]
+\caption{Baseline Model Performance}
+\label{tab:baseline_results}
+\begin{tabular}{lcccccccccc}
```
Comment (VladTemp27) [inline paper/apa/tables/baseline_results.tex]: Header row at line 7 has 11 columns (Model, Sampler, Accuracy, ROC-AUC, Precision (Fail), Recall (Fail), F1 (Fail), Precision (Succ), Recall (Succ), F1 (Succ), Train Time (s)) but the column format only declares 10 (l + 9×c). LaTeX will fail to compile.
Fix: Change to \begin{tabular}{lccccccccccc} (add one more c).

File: non-temporal/run_experiments.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    
+    for v_name, feat_ver, samplers in versions:
+        print(f"\n--- Running {v_name} Experiments ---")
+        features = pipeline.get_features(feat_ver)
+        X = df[features]
+        y = df['Target']
+        
+        # 70/20/10 split as per the notebooks
+        X_temp, X_val, y_temp, y_val = train_test_split(X, y, test_size=0.10, random_state=42, stratify=y)
+        X_train, X_test, y_train, y_test = train_test_split(X_temp, y_temp, test_size=2/9, random_state=42, stratify=y_temp)
```
Comment (VladTemp27) [inline non-temporal/run_experiments.py:44]: X_val and y_val are created but never referenced again. This suggests an incomplete implementation — final holdout evaluation is missing.

File: paper/apa/thesis_apa.tex [hunk 4]
```diff
@@ -23,14 +27,12 @@
...
-% ----- References (APA-like with natbib) -----
-% NOTE: For strict APA 7th, install apacite and uncomment:
-%   \usepackage[natbibapa]{apacite}
-%   \bibliographystyle{apacite}
-\usepackage[round]{natbib}
-\bibliographystyle{apalike}
+% ----- References: apacite must be loaded BEFORE hyperref -----
+\usepackage[natbibapa]{apacite}
+\bibliographystyle{apacite}
+\shortcites{ref3,ref5,ref7,ref8,ref9,ref11,ref12,ref13,ref14,ref15,ref16,ref17,ref21,ref22,ref23,ref24,ref27,ref30}
```
Comment (VladTemp27) [inline paper/apa/thesis_apa.tex]: Several references cited in the body (ref2, ref4, ref6, ref18, ref19, ref25, ref26, ref28, ref29) are missing from \shortcites, causing inconsistent citation formatting between parenthetical and inline styles.

File: non-temporal/run_experiments.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
+import logging
+logging.getLogger().setLevel(logging.CRITICAL)
```
Comment (VladTemp27) [inline non-temporal/run_experiments.py]: Suppressing the root logger globally at CRITICAL hides errors from all libraries (sklearn, pandas, xgboost). The UserWarning filter at line 16 (experiment_pipeline.py) may mask real feature-name mismatch issues. This is the same for experiment_pipeline

File: non-temporal/experiment_pipeline.py [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+        FAILURE_OUTCOMES = {'DIED', 'LOST TO FF-UP', 'FAILED'}
+
+        df = df_raw[df_raw['Outcome_Label'].isin(SUCCESS_OUTCOMES | FAILURE_OUTCOMES)].copy()
+        df['Target'] = df['Outcome_Label'].apply(lambda x: 1 if x in SUCCESS_OUTCOMES else 0)
+        
+        # Derived features (if possible)
+        if 'Date Started Tx' in df.columns and 'Date of Diagnosis' in df.columns:
+            df['_Date_Started_Tx'] = pd.to_datetime(df['Date Started Tx'], errors='coerce')
+            df['_Date_of_Diagnosis'] = pd.to_datetime(df['Date of Diagnosis'], errors='coerce')
+            df['Diagnosis_to_Treatment_days'] = (df['_Date_Started_Tx'] - df['_Date_of_Diagnosis']).dt.days.clip(lower=0)
```
Comment (VladTemp27) [inline non-temporal/experiment_pipeline.py]: Feature is derived during load_data but neither baseline nor improved feature lists (lines 74-81) reference it. Dead code.

**Review by VladTemp27 — CHANGES_REQUESTED** [review]

File: non-temporal/experiment_pipeline.py (hunk 1 above)
Comment (Benny-Gil) [reply non-temporal/experiment_pipeline.py:64]: Added `assesrt` that encoded count matches `OUTCOME_LABELS_ALPHA` length, with a descriptive error message.

File: paper/apa/tables/baseline_results.tex (hunk 2 above)
Comment (Benny-Gil) [reply paper/apa/tables/baseline_results.tex]: added one c

File: non-temporal/run_experiments.py (hunk 3 above)
Comment (Benny-Gil) [reply non-temporal/run_experiments.py:44]: store val splits in val_sets dict per version, tagged each result with `Version`, and added final evaluation (ROC-AUC + Accuracy) of the best model after training

File: paper/apa/thesis_apa.tex (hunk 4 above)
Comment (Benny-Gil) [reply paper/apa/thesis_apa.tex]: added missing references to \shortcites list

File: non-temporal/experiment_pipeline.py (hunk 6 above)
Comment (Benny-Gil) [reply non-temporal/experiment_pipeline.py]: ded code removed

File: non-temporal/run_experiments.py (hunk 5 above)
Comment (Benny-Gil) [reply non-temporal/run_experiments.py]: logger is now verbose, print statements replaced with logger

**Review by VladTemp27 — APPROVED** [review]
